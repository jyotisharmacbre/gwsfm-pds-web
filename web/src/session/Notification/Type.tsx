import { Action } from 'redux';

export interface INotification {
  notificationCount: number;
}

export interface IGetNotificationBeginAction
  extends Action<'GetNotificationBeginAction'> {}

export interface IGetNotificationSuccessAction
  extends Action<'GetNotificationSuccessAction'> {
  data: INotification;
}

export interface IGetNotificationFailureAction
  extends Action<'GetNotificationFailureAction'> {
  error: any;
}

export type NotificationActions =
  | IGetNotificationBeginAction
  | IGetNotificationSuccessAction
  | IGetNotificationFailureAction;
