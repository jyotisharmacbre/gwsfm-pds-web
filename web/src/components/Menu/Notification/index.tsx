import React, { useEffect } from 'react';
import { INotification } from "../../../store/Notifications/Types/INotification";
import IReactIntl from "../../../Translations/IReactIntl";
import { injectIntl } from "react-intl";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBell } from "@fortawesome/free-solid-svg-icons";
import { useHistory } from "react-router-dom";
import { ILookup } from '../../../store/Lookups/Types/ILookup';


interface IMapStateToProps {
	notifications: Array<INotification>;
	showNotification: boolean;
	lookups: Array<ILookup>;
}


const Notification: React.FC<IMapStateToProps & IReactIntl> = (props) => {
	let history = useHistory();

	//add & remove class for pipeline and dashboard page
	const showClass = () => {
		return history.location.pathname == "/" ||
			history.location.pathname == "/Pipeline"
	}

	const getTemplate = (notificationType) => {
		return props.lookups?.find(x => x.lookupKey == notificationType && x.lookupItem == 'Notification_Template')?.description;
	}

	const getUnreadNotificationCount = () => {
		return props.notifications?.filter(x => x.status === 0).length;
	}

	return (
		<div>
			<i>
				<FontAwesomeIcon className="" icon={faBell} />
				{getUnreadNotificationCount() > 0 && <span className="badge badge-light"></span>}
			</i>
			{getUnreadNotificationCount()>0 ? <div className={`dropdown-menu notify_dropdown user-dropdown ${props.showNotification ? 'show' : 'hide'}
	  ${showClass() ? 'default' : 'all_pages'} `}>
				<ul>
					<li>
						<div className="notify_topbar">
							<span>({getUnreadNotificationCount()}) New Notifications</span>
							<span>Mark all as Read</span>
						</div>
					</li>
					{props.notifications.map(notification => {
						return (<li className="bg-grey">
							<a href={`/ReviewApprove/${notification.projectId}`}>
							<h4 className="title">
								{`${getTemplate(notification.notificationType)}`}
							</h4>
							<p className="brief">
								{notification.description}
							</p>
							</a>
							<span className="mark_sign">MARK READ</span>
						</li>)

					})}
				</ul>
			</div>
			:
     <div className={`dropdown-menu notify_dropdown user-dropdown ${props.showNotification ? 'show' : 'hide'}
	  ${showClass() ? 'default' : 'all_pages'} `}>
				<ul>
					<li>
						<p>No Notification for you</p>
					</li>
				</ul>
			</div>}
		</div>

	);
};


export default injectIntl(Notification);
