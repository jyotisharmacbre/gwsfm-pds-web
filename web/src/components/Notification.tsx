import React from 'react';
import NotificationsIconImage from '@material-ui/icons/Notifications';
import { IconButton, Badge } from '@material-ui/core';

export default function Notification(props: {NotificationCount:number, handleClick: any}){
   
        return (
           <IconButton color="inherit" onClick={props.handleClick}>
            <Badge badgeContent={props.NotificationCount}  > 
              <NotificationsIconImage id="BellIcon" name="BellIcon" color="secondary"  />
            </Badge>
          </IconButton>
         
          );
   
   
  }
  