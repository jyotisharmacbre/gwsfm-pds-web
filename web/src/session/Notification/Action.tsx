import { ActionCreator, Dispatch } from "redux";
import { ThunkAction } from "redux-thunk";
import { INotification, IGetNotificationSuccessAction, IGetNotificationBeginAction, IGetNotificationFailureAction } from "./Type";

export const getNotificationActionCreator: ActionCreator<ThunkAction<Promise<IGetNotificationSuccessAction>, INotification, null, IGetNotificationSuccessAction>> = () => {
    return async (dispatch: Dispatch) => {
        const beginAction: IGetNotificationBeginAction = {
            type: 'GetNotificationBeginAction'
        };

        dispatch(beginAction);
        
        const sucessAction: IGetNotificationSuccessAction = {
            type: 'GetNotificationSuccessAction',
            data: {notificationCount: 5},
        };

        return dispatch(sucessAction)

    }
}