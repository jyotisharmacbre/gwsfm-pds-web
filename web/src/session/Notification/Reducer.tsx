import { NotificationActions } from './Type';
import { INotificationState } from '../state';
import { Reducer } from 'redux';

const initialState: INotificationState = {
  notificationCount: 0
};

const notificationReducer: Reducer<INotificationState, NotificationActions> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case 'GetNotificationBeginAction':
      return { ...state };
    case 'GetNotificationSuccessAction':
      return Object.assign({}, state, action.data);
    //  case 'GetNotificationFailureAction':

    default:
      return state;
  }
};

export default notificationReducer;
