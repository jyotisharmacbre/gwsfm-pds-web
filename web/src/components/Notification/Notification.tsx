import React, { useEffect } from 'react';
import NotificationsIconImage from '@material-ui/icons/Notifications';
import { IconButton, Badge } from '@material-ui/core';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { connect } from 'react-redux';
import { IApplicationState } from '../../session/rootReducer';
import {  INotificationProps } from '../../props/AppProps';
import { getNotificationActionCreator } from '../../session/Notification/Action';

const mapDispatchToProps = (dispatch: ThunkDispatch<any, any, AnyAction>) => {
  return {
    handleClick: () => { window.location.href = "/Notifications"; },
    getNotificationCount: () => dispatch(getNotificationActionCreator())
  }
};

const mapStateToProps = (state: IApplicationState) => {
  return {
    data: state.notificationState
  }
};

const Notification: React.FC<INotificationProps> = (props) => {

  useEffect(() => {
    props.getNotificationCount();
  }, []);

  return (
    <IconButton color="inherit" onClick={props.handleClick}>
      <Badge badgeContent="5" >
        <NotificationsIconImage id="BellIcon" name="BellIcon" color="secondary" />
        
      </Badge>
    </IconButton>
  );


}

export default connect(mapStateToProps, mapDispatchToProps)(Notification);
