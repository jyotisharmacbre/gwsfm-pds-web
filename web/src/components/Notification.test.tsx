import React from 'react';
import ReactDOM from 'react-dom';
import Notification from './Notification';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

it('Notification should render', () => {
    const component = renderer.create(
        <Notification NotificationCount={5} handleClick={[]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

 it('Notification should render icon', () => {
   const props ={
    NotificationCount: 4,
    handleClick:() => {
    }
   }
     const component = mount(
        <Notification {...props} />
    );

    expect(component.find('#BellIcon').exists());  
 });

 
