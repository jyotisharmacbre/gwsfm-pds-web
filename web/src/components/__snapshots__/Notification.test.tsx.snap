// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      className="MuiBadge-root"
      onClick={Array []}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
        focusable="false"
        id="BellIcon"
        name="BellIcon"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
        />
      </svg>
      <span
        className="MuiBadge-badge"
      >
        5
      </span>
    </span>
  </span>
</button>
`;
